cmd/src/debug_comp.go

```diff
@@ -0,0 +1,255 @@
+		if verified, _ := verify("Do you want to start writing to an archive?"); !verified {
+			return nil
+		}
+
+		err = archiveCompose(ctx, zw, *verbose, configs, baseDir)
+		if err != nil {
+			return cmderrors.ExitCode(1, nil)
+		}
+		return nil
+	}
+
+	debugCommands = append(debugCommands, &command{
+		flagSet: flagSet,
+		handler: handler,
+		usageFunc: func() {
+			fmt.Println(usage)
+		},
+	})
+}
+
+/*
+Docker functions
+*/
+
+func archiveCompose(ctx context.Context, zw *zip.Writer, verbose, configs bool, baseDir string) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	containers, err := getContainers(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get docker containers: %w", err)
+	}
+
+	if verbose {
+		log.Printf("getting docker data for %d containers...\n", len(containers))
+	}
+
+	// setup channel for slice of archive function outputs
+	ch := make(chan *archiveFile)
+	g, ctx := errgroup.WithContext(ctx)
+	semaphore := semaphore.NewWeighted(8)
+
+	// start goroutine to run docker ps -o wide
+	g.Go(func() error {
+		if err := semaphore.Acquire(ctx, 1); err != nil {
+			return err
+		}
+		defer semaphore.Release(1)
+		ch <- getPs(ctx, baseDir)
+		return nil
+	})
+
+	// start goroutine to run docker container stats --no-stream
+	g.Go(func() error {
+		if err := semaphore.Acquire(ctx, 1); err != nil {
+			return err
+		}
+		defer semaphore.Release(1)
+		ch <- getStats(ctx, baseDir)
+		return nil
+	})
+
+	// start goroutine to run docker container logs <container>
+	for _, container := range containers {
+		c := container
+		g.Go(func() error {
+			if err := semaphore.Acquire(ctx, 1); err != nil {
+				return err
+			}
+			defer semaphore.Release(1)
+			ch <- getContainerLog(ctx, c, baseDir)
+			return nil
+		})
+	}
+
+	// start goroutine to run docker container inspect <container>
+	for _, container := range containers {
+		c := container
+		g.Go(func() error {
+			if err := semaphore.Acquire(ctx, 1); err != nil {
+				return err
+			}
+			defer semaphore.Release(1)
+			ch <- getInspect(ctx, c, baseDir)
+			return nil
+		})
+	}
+
+	// start goroutine to get configs
+	if configs {
+		g.Go(func() error {
+			if err := semaphore.Acquire(ctx, 1); err != nil {
+				return err
+			}
+			defer semaphore.Release(1)
+			ch <- getExternalServicesConfig(ctx, baseDir)
+			return nil
+		})
+
+		g.Go(func() error {
+			if err := semaphore.Acquire(ctx, 1); err != nil {
+				return err
+			}
+			defer semaphore.Release(1)
+			ch <- getSiteConfig(ctx, baseDir)
+			return nil
+		})
+	}
+
+	// close channel when wait group goroutines have completed
+	go func() {
+		g.Wait()
+		close(ch)
+	}()
+
+	// Read binaries from channel and write to archive on host machine
+	if err := writeChannelContentsToZip(zw, ch, verbose); err != nil {
+		return fmt.Errorf("failed to write archives from channel: %w", err)
```

Comment:
```suggestion
		return errors.Wrap(err, "failed to write archives from channel")
```

This is the idiomatic way to wrap errors (please check other uses). We try to only use our internal `errors` package and not fmt/other error package functions for error handling/construction (not all libraries interact nicely together). This is less of an issue for src-cli, but a good thing to mention in this eng org.